java/java.source/src/org/netbeans/modules/java/ui/Icons.java

```diff
@@ -44,64 +41,59 @@
     private Icons() {
     }
     
-    public static Icon getBusyIcon () {
-        Image img = ImageUtilities.loadImage (WAIT);
-        if (img == null) {
-            return null;
-        }
-        else {
-            return ImageUtilities.image2Icon (img);
-        }
+    public static ImageIcon getBusyIcon () {
+        ImageIcon icon = ImageUtilities.loadImageIcon(WAIT, false);
+        return icon;
     }
             
     
-    public static Icon getElementIcon( ElementKind elementKind, Collection<Modifier> modifiers ) {
+    public static ImageIcon getElementIcon( ElementKind elementKind, Collection<Modifier> modifiers ) {
```

Comment:
You can also remove the cast and "// todo: what happens, if i get a CCE here?" comment at https://github.com/apache/netbeans/blob/master/groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/api/completion/CompletionItem.java#L1006 .

Reply:
That change would require an API change and sigtest fails on that one. I'd left that out right now.